### PR TITLE
Feat: analysis radical migration tool

### PR DESCRIPTION
**Pull request by ehhartmann**
this tool will create a json file of migrations which can be visualized in pymol using an external script

### CODE REVIEW

File: src/kimmdy/analysis.py [hunk 1]
```diff
@@ -350,6 +351,86 @@ def radical_population(
         run_shell_cmd(f"vmd {pdb_output}", cwd=analysis_dir)
 
 
+def radical_migration(
+    dir: str,
```
Comment (jmbuhr) [inline src/kimmdy/analysis.py:355]: looks like dir is not a string but rather a list of strings. I would rename to `dirs` and change the typing accordingly.

File: src/kimmdy/analysis.py [hunk 2]
```diff
@@ -350,6 +351,86 @@ def radical_population(
...
+
+            from_atom = [
+                key for key, value in connectivity_difference.items() if value == 1
+            ]
+            to_atom = [
+                key for key, value in connectivity_difference.items() if value == -1
+            ]
+            if len(from_atom) == 1 and len(to_atom) == 1:
+                migrations.append([from_atom[0], to_atom[0], max(sorted_recipe.rates)])
+    print(migrations)
```
Comment (jmbuhr) [inline src/kimmdy/analysis.py:415]: leftover print (or needs user-facing explanation)

File: src/kimmdy/analysis.py [hunk 3]
```diff
@@ -643,7 +724,30 @@ def get_analysis_cmdline_args() -> argparse.Namespace:
         help="Open VMD with the concatenated trajectory."
         "To view the radical occupancy per atom, add a representation with the beta factor as color.",
     )
-
+    parser_radical_migration = subparsers.add_parser(
+        name="radical_migration",
+        help="Create a csv of radical migration events for further analysis.",
```
Comment (jmbuhr) [inline src/kimmdy/analysis.py:729]: json, not csv, right?

File: src/kimmdy/assets/analyse_kimmdy_migration.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import MDAnalysis as mda
```
Comment (jmbuhr) [inline src/kimmdy/assets/analyse_kimmdy_migration.py:1]: maybe `assets` should be a `scripts` folder (or scripts is a subdirectory of assets)

**Review by jmbuhr — CHANGES_REQUESTED** [review]

**Review by jmbuhr — COMMENTED** [review]
How do you want users to run the script? Right now it would be something like `python -m kimmdy.assets.scripts.analyse_kimmdy_migration.py`, in which case you would want to add the classic `if __name__ == "__main__"` thing. Though from the `# %%` comments it looks more like something you expect people to run interactively in their own environment, so I'm wondering if some place in the `guide` docs folder would be a better solution, to separate it from the regular kimmdy modules. Also, do you want to add `pymol` to https://github.com/hits-mbm-dev/kimmdy/blob/0bc5a68b7adbd026a7a7cd78176a2d47b602dc68/setup.cfg#L60-L72 ?

**Comment by ehhartmann** [comment]
I would say installing pymol is a bit messy, so I would leave it to the user. 
The script runs with `pymol <script_name>`. For debugging and adding new features I run it interactively. To me, it is just a nice script to share with people who have kimmdy output. We could put it in guide and just have it as asset for some future tutorial

**Review by jmbuhr — APPROVED** [review]

**Pull request merged**